lib/trace-agent.js

```diff
@@ -134,6 +133,9 @@ TraceAgent.prototype.runInSpan = function(name, labels, fn) {
  */
 TraceAgent.prototype.setTransactionName = function(name) {
   var rootSpan = cls.getRootContext();
+  if (rootSpan === SpanData.nullSpan) {
```

Comment:
We can push the name access down into `SpanData` but to remove this check from `addContextToHeaders`, we would need to push the context logic down as well. Since the accesses to `SpanData.span` are isolated to `TraceAgent` I think we're ok for now.